### PR TITLE
[8.x] Add nested relationships to whereRelation function

### PR DESCRIPTION
**Pull request by josemariagomez**
This PR added the possibility to use nested relationships when using 'whereRelation' Eloquent's function, using dot notation.
Also still works with not nested relationships, so it's not a breaking change.

## How it works
```php
// Before
Company::whereHas('teams', function($query) {
  $query->whereHas('users', function($query) {
      $query->whereHas('roles', function($query) {
          $query->where('name', 'admin');
      });
  });
})->get();

// After 💅🏻
Company::whereRelation('teams.users.roles', 'name', 'admin')->get();

// Also can be used as orWhereRelation
Company::where('public', true)->orWhereRelation('teams.users', 'id', auth()->id())->get();

// Still can use this function for not nested relationships
Company::whereRelation('teams', 'id', '1')->get();
```

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php [hunk 1]
```diff
@@ -353,6 +353,7 @@ public function orWhereDoesntHaveMorph($relation, $types, Closure $callback = nu
 
     /**
      * Add a basic where clause to a relationship query.
+     * Relationship can be nested using dot notation.
```
Comment (driesvints) [inline src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php:356]: Remove this second line again.

File: src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php [hunk 2]
```diff
@@ -362,13 +363,26 @@ public function orWhereDoesntHaveMorph($relation, $types, Closure $callback = nu
...
+                    $relations->shift();
+                    $query->whereRelation($relations->implode('.'), $column, $operator, $value);
+                });
+            }
+        );
     }
 
     /**
      * Add an "or where" clause to a relationship query.
+     * Relationship can be nested using dot notation.
```
Comment (driesvints) [inline src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php:385]: Remove this second line again.

File: src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php [hunk 3]
```diff
@@ -362,13 +363,26 @@ public function orWhereDoesntHaveMorph($relation, $types, Closure $callback = nu
      */
     public function whereRelation($relation, $column, $operator = null, $value = null)
     {
-        return $this->whereHas($relation, function ($query) use ($column, $operator, $value) {
-            $query->where($column, $operator, $value);
-        });
+        $relations = collect(explode('.', $relation));
+        return $this->when(
```
Comment (driesvints) [inline src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php:367]:
```suggestion
        $relations = collect(explode('.', $relation));

        return $this->when(
```

File: src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php [hunk 4]
```diff
@@ -378,9 +392,21 @@ public function whereRelation($relation, $column, $operator = null, $value = nul
      */
     public function orWhereRelation($relation, $column, $operator = null, $value = null)
     {
-        return $this->orWhereHas($relation, function ($query) use ($column, $operator, $value) {
-            $query->where($column, $operator, $value);
-        });
+        $relations = collect(explode('.', $relation));
+        return $this->when(
```
Comment (driesvints) [inline src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php:396]:
```suggestion
        $relations = collect(explode('.', $relation));

        return $this->when(
```

**Review by driesvints — CHANGES_REQUESTED** [review]

**Comment by devfrey** [comment]
Why was this merged?

1) It was already possible to used nested relations before these code changes.
2) This PR contains no tests...

**Pull request merged**